quixstreams/platforms/quix/api.py

```diff
@@ -63,24 +61,47 @@ def request(self, method, url, **kwargs):
     @property
     def default_workspace_id(self) -> str:
         if not self._default_workspace_id:
-            raise self.UndefinedQuixWorkspaceId("You must provide a Quix Workspace ID")
+            raise UndefinedQuixWorkspaceId("You must provide a Quix Workspace ID")
         return self._default_workspace_id
 
     @default_workspace_id.setter
     def default_workspace_id(self, value):
         self._default_workspace_id = value
 
+    def _response_handler(self, r: requests.Response, *args, **kwargs):
+        """
+        Custom callback/hook that is called after receiving a request.Response
+
+        Catches non-200's and passes both the original exception and the Response body.
+
+        Note: *args and **kwargs expected for hook
+        """
+        try:
+            r.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            try:
+                reason_text = e.response.json()
+            except requests.exceptions.JSONDecodeError:
+                reason_text = e.response.text
+
+            raise QuixApiRequestFailure(
+                {
+                    "response_error": e,  # response error
+                    "response_body": reason_text,  # error message in the API response
+                }
+            )
+
     def _init_session(self) -> SessionWithUrlBase:
         if not (self._portal_api and self._auth_token):
             missing = filter(
                 None,
                 [self._portal_api or "portal_api", self._auth_token or "auth_token"],
             )
```

Comment:
Since we're touching the code nearby, this block has issues too.
`filter()` creates a lazy object, and when you cast it to `str`, you get `'<filter object at 0x000000>'`

I suggest we remove these checks from `_init_session` and instead do them in `QuixPortalApiService.__init__`

Plus, we don't need to check portal API anymore - there's always a default value.
We only need to check the `auth_token`, and `QuixPortalApiService.__init__` is the best place to do that